### DISCUSSION TITLE
Ability to augment the error message of rules

### DISCUSSION THREAD

**Discussion by mrcljx** [astral-sh/ruff#11635] [Ideas]
We use E711 and E712 which tell the user that `== None` should be `is None`. The problem is that there's an exception to that rule: when using SQLAlchemy one MUST use `.filter(column == None)` so that `column.__eq__` is called.

It would be great if we could provide some rationale/additional info in Ruff error messages, with something like:

```
[rules]
E711.rationale = ""
```

**Comment by Mulugruntz** [comment]
Even better if it could not flag it as E711, as it's the correct syntax.

**Comment by Apakottur** [comment]
Is it possible to use `.filter(column.is_(None))` in your case? Maybe that avoids the rule whatsoever
